### ISSUE TITLE
Rule proposal: autocomplete

### ISSUE THREAD

**Issue by lencioni** [evcohen/eslint-plugin-jsx-a11y#44]
I'm not entirely sure what this rule could do, or even if it is a11y related, but it crossed my mind and I figured I'd open an issue to discuss it.

HTML has an autocomplete attribute that can be used to provide powerful form autofilling features. Here's the spec: https://html.spec.whatwg.org/multipage/forms.html#autofill

It would be useful to have a rule that would at least validate the attribute values, but perhaps there is more that can be done?

**Comment by evcohen** [comment]
that would be turning this into a static type checker 😝 can you not use React PropTypes for that? if we can find an a11y use case then for sure this fits in scope, or maybe we can branch off a create a new plugin like `eslint-plugin-jsx-html` that'll statically validate HTML rules and such against JSX.

**Comment by lencioni** [comment]
> can you not use React PropTypes for that?

I think that would require defining proptypes on DOM elements, since these aren't components. There might be a way to do that, but I haven't come across it.

> if we can find an a11y use case then for sure this fits in scope, or maybe we can branch off a create a new plugin like `eslint-plugin-jsx-html` that'll statically validate HTML rules and such against JSX.

I think that the basic value checking by itself is outside the scope of this project. I think that proactively suggesting `autocomplete` props or requiring an `autocomplete` prop of some kind might be in the scope of this project--at which point it might make sense to also add the type checking.

**Comment by evcohen** [comment]
any correlation https://www.w3.org/TR/wai-aria/complete#aria-autocomplete ?

**Comment by lencioni** [comment]
No, I think that's for interfaces that allow tagging with autocomplete in textboxes while typing, like what happens on github when you type `@` or `#`.

**Comment by jessebeach** [comment]
FYI, `aria-autocomplete` was rewritten in ARIA 1.1 by @mcking65. It's now much more aligned with a concept of typeahead.
